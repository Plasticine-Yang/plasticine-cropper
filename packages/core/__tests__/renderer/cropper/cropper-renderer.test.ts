import { CropperRenderer, CropperRendererImpl } from '@internal-dist/index'

describe('cropper renderer', () => {
  const setup = () => {
    const parentNodeOfImageElement = document.createElement('div')
    const imageElement = document.createElement('img')

    parentNodeOfImageElement.appendChild(imageElement)

    const cropperRenderer: CropperRenderer = new CropperRendererImpl(imageElement)

    return {
      parentNodeOfImageElement,
      cropperRenderer,
      imageElement,
    }
  }

  test('should throw error when no container of image', () => {
    const imageElement = document.createElement('img')

    expect(() => {
      new CropperRendererImpl(imageElement)
    }).toThrowError('图片缺少容器元素')
  })

  test('should mount', () => {
    const { parentNodeOfImageElement, cropperRenderer } = setup()

    cropperRenderer.mount()

    expect(parentNodeOfImageElement).toMatchSnapshot()
  })

  test('should unmount', () => {
    const { parentNodeOfImageElement, cropperRenderer } = setup()

    cropperRenderer.mount()

    expect(parentNodeOfImageElement).toMatchSnapshot()

    cropperRenderer.unmount()

    expect(parentNodeOfImageElement).toMatchSnapshot()
  })

  test('should get all cropper elements', () => {
    const { cropperRenderer } = setup()

    cropperRenderer.mount()

    expect(cropperRenderer.getCropperElements()).toMatchSnapshot()

    cropperRenderer.unmount()

    expect(cropperRenderer.getCropperElements()).toBeNull()
  })

  test('should toggle visibility of raw image element', () => {
    const { cropperRenderer, imageElement } = setup()

    cropperRenderer.mount()

    // hide
    cropperRenderer.hideRawImageElement()
    expect(Array.from(imageElement.classList)).toMatchInlineSnapshot(`
      [
        "plasticine-cropper-common-hidden",
      ]
    `)

    // show
    cropperRenderer.showRawImageElement()
    expect(Array.from(imageElement.classList)).toMatchInlineSnapshot('[]')
  })
})
