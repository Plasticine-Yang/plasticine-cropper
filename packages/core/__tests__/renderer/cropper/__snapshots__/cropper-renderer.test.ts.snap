// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cropper renderer > should get all cropper elements 1`] = `
{
  "cropContainer": <div
    class="plasticine-cropper__crop-container"
  >
    <div
      class="plasticine-cropper__view-container"
    />
    <div
      class="plasticine-cropper__dotted-line plasticine-cropper__dotted-line--horizontal"
    />
    <div
      class="plasticine-cropper__dotted-line plasticine-cropper__dotted-line--vertical"
    />
    <div
      class="plasticine-cropper__move-placeholder"
    />
    <div
      class="plasticine-cropper__focus"
    />
    <div
      class="plasticine-cropper__line plasticine-cropper__line--n"
      data-plasticine-cropper-resize-direction="n"
    />
    <div
      class="plasticine-cropper__line plasticine-cropper__line--e"
      data-plasticine-cropper-resize-direction="e"
    />
    <div
      class="plasticine-cropper__line plasticine-cropper__line--s"
      data-plasticine-cropper-resize-direction="s"
    />
    <div
      class="plasticine-cropper__line plasticine-cropper__line--w"
      data-plasticine-cropper-resize-direction="w"
    />
    <div
      class="plasticine-cropper__point plasticine-cropper__point--n"
      data-plasticine-cropper-resize-direction="n"
    />
    <div
      class="plasticine-cropper__point plasticine-cropper__point--e"
      data-plasticine-cropper-resize-direction="e"
    />
    <div
      class="plasticine-cropper__point plasticine-cropper__point--s"
      data-plasticine-cropper-resize-direction="s"
    />
    <div
      class="plasticine-cropper__point plasticine-cropper__point--w"
      data-plasticine-cropper-resize-direction="w"
    />
    <div
      class="plasticine-cropper__point plasticine-cropper__point--ne"
      data-plasticine-cropper-resize-direction="ne"
    />
    <div
      class="plasticine-cropper__point plasticine-cropper__point--nw"
      data-plasticine-cropper-resize-direction="nw"
    />
    <div
      class="plasticine-cropper__point plasticine-cropper__point--sw"
      data-plasticine-cropper-resize-direction="sw"
    />
    <div
      class="plasticine-cropper__point plasticine-cropper__point--se"
      data-plasticine-cropper-resize-direction="se"
    />
  </div>,
  "cropContainerLines": {
    "e": <div
      class="plasticine-cropper__line plasticine-cropper__line--e"
      data-plasticine-cropper-resize-direction="e"
    />,
    "n": <div
      class="plasticine-cropper__line plasticine-cropper__line--n"
      data-plasticine-cropper-resize-direction="n"
    />,
    "s": <div
      class="plasticine-cropper__line plasticine-cropper__line--s"
      data-plasticine-cropper-resize-direction="s"
    />,
    "w": <div
      class="plasticine-cropper__line plasticine-cropper__line--w"
      data-plasticine-cropper-resize-direction="w"
    />,
  },
  "cropContainerMovePlaceholder": <div
    class="plasticine-cropper__move-placeholder"
  />,
  "cropContainerPoints": {
    "e": <div
      class="plasticine-cropper__point plasticine-cropper__point--e"
      data-plasticine-cropper-resize-direction="e"
    />,
    "n": <div
      class="plasticine-cropper__point plasticine-cropper__point--n"
      data-plasticine-cropper-resize-direction="n"
    />,
    "ne": <div
      class="plasticine-cropper__point plasticine-cropper__point--ne"
      data-plasticine-cropper-resize-direction="ne"
    />,
    "nw": <div
      class="plasticine-cropper__point plasticine-cropper__point--nw"
      data-plasticine-cropper-resize-direction="nw"
    />,
    "s": <div
      class="plasticine-cropper__point plasticine-cropper__point--s"
      data-plasticine-cropper-resize-direction="s"
    />,
    "se": <div
      class="plasticine-cropper__point plasticine-cropper__point--se"
      data-plasticine-cropper-resize-direction="se"
    />,
    "sw": <div
      class="plasticine-cropper__point plasticine-cropper__point--sw"
      data-plasticine-cropper-resize-direction="sw"
    />,
    "w": <div
      class="plasticine-cropper__point plasticine-cropper__point--w"
      data-plasticine-cropper-resize-direction="w"
    />,
  },
  "root": <div
    class="plasticine-cropper"
  >
    <div
      class="plasticine-cropper__overlay"
    />
    <div
      class="plasticine-cropper__canvas"
    />
    <div
      class="plasticine-cropper__crop-container"
    >
      <div
        class="plasticine-cropper__view-container"
      />
      <div
        class="plasticine-cropper__dotted-line plasticine-cropper__dotted-line--horizontal"
      />
      <div
        class="plasticine-cropper__dotted-line plasticine-cropper__dotted-line--vertical"
      />
      <div
        class="plasticine-cropper__move-placeholder"
      />
      <div
        class="plasticine-cropper__focus"
      />
      <div
        class="plasticine-cropper__line plasticine-cropper__line--n"
        data-plasticine-cropper-resize-direction="n"
      />
      <div
        class="plasticine-cropper__line plasticine-cropper__line--e"
        data-plasticine-cropper-resize-direction="e"
      />
      <div
        class="plasticine-cropper__line plasticine-cropper__line--s"
        data-plasticine-cropper-resize-direction="s"
      />
      <div
        class="plasticine-cropper__line plasticine-cropper__line--w"
        data-plasticine-cropper-resize-direction="w"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--n"
        data-plasticine-cropper-resize-direction="n"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--e"
        data-plasticine-cropper-resize-direction="e"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--s"
        data-plasticine-cropper-resize-direction="s"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--w"
        data-plasticine-cropper-resize-direction="w"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--ne"
        data-plasticine-cropper-resize-direction="ne"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--nw"
        data-plasticine-cropper-resize-direction="nw"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--sw"
        data-plasticine-cropper-resize-direction="sw"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--se"
        data-plasticine-cropper-resize-direction="se"
      />
    </div>
  </div>,
}
`;

exports[`cropper renderer > should mount 1`] = `
<div>
  <img />
  <div
    class="plasticine-cropper"
  >
    <div
      class="plasticine-cropper__overlay"
    />
    <div
      class="plasticine-cropper__canvas"
    />
    <div
      class="plasticine-cropper__crop-container"
    >
      <div
        class="plasticine-cropper__view-container"
      />
      <div
        class="plasticine-cropper__dotted-line plasticine-cropper__dotted-line--horizontal"
      />
      <div
        class="plasticine-cropper__dotted-line plasticine-cropper__dotted-line--vertical"
      />
      <div
        class="plasticine-cropper__move-placeholder"
      />
      <div
        class="plasticine-cropper__focus"
      />
      <div
        class="plasticine-cropper__line plasticine-cropper__line--n"
        data-plasticine-cropper-resize-direction="n"
      />
      <div
        class="plasticine-cropper__line plasticine-cropper__line--e"
        data-plasticine-cropper-resize-direction="e"
      />
      <div
        class="plasticine-cropper__line plasticine-cropper__line--s"
        data-plasticine-cropper-resize-direction="s"
      />
      <div
        class="plasticine-cropper__line plasticine-cropper__line--w"
        data-plasticine-cropper-resize-direction="w"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--n"
        data-plasticine-cropper-resize-direction="n"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--e"
        data-plasticine-cropper-resize-direction="e"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--s"
        data-plasticine-cropper-resize-direction="s"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--w"
        data-plasticine-cropper-resize-direction="w"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--ne"
        data-plasticine-cropper-resize-direction="ne"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--nw"
        data-plasticine-cropper-resize-direction="nw"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--sw"
        data-plasticine-cropper-resize-direction="sw"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--se"
        data-plasticine-cropper-resize-direction="se"
      />
    </div>
  </div>
</div>
`;

exports[`cropper renderer > should unmount 1`] = `
<div>
  <img />
  <div
    class="plasticine-cropper"
  >
    <div
      class="plasticine-cropper__overlay"
    />
    <div
      class="plasticine-cropper__canvas"
    />
    <div
      class="plasticine-cropper__crop-container"
    >
      <div
        class="plasticine-cropper__view-container"
      />
      <div
        class="plasticine-cropper__dotted-line plasticine-cropper__dotted-line--horizontal"
      />
      <div
        class="plasticine-cropper__dotted-line plasticine-cropper__dotted-line--vertical"
      />
      <div
        class="plasticine-cropper__move-placeholder"
      />
      <div
        class="plasticine-cropper__focus"
      />
      <div
        class="plasticine-cropper__line plasticine-cropper__line--n"
        data-plasticine-cropper-resize-direction="n"
      />
      <div
        class="plasticine-cropper__line plasticine-cropper__line--e"
        data-plasticine-cropper-resize-direction="e"
      />
      <div
        class="plasticine-cropper__line plasticine-cropper__line--s"
        data-plasticine-cropper-resize-direction="s"
      />
      <div
        class="plasticine-cropper__line plasticine-cropper__line--w"
        data-plasticine-cropper-resize-direction="w"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--n"
        data-plasticine-cropper-resize-direction="n"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--e"
        data-plasticine-cropper-resize-direction="e"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--s"
        data-plasticine-cropper-resize-direction="s"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--w"
        data-plasticine-cropper-resize-direction="w"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--ne"
        data-plasticine-cropper-resize-direction="ne"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--nw"
        data-plasticine-cropper-resize-direction="nw"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--sw"
        data-plasticine-cropper-resize-direction="sw"
      />
      <div
        class="plasticine-cropper__point plasticine-cropper__point--se"
        data-plasticine-cropper-resize-direction="se"
      />
    </div>
  </div>
</div>
`;

exports[`cropper renderer > should unmount 2`] = `
<div>
  <img />
</div>
`;
